arangod/Cluster/AgencyCache.cpp

```diff
@@ -256,10 +337,19 @@ void AgencyCache::run() {
                         std::lock_guard g(_storeLock);
                         _readDB.applyTransaction(i); // apply logs
                         _commitIndex = i.get("index").getNumber<uint64_t>();
-                      }
-                      {
-                        std::lock_guard g(_callbacksLock);
-                        handleCallbacksNoLock(i.get("query"), uniq, toCall);
+
+                        std::unordered_set<std::string> pc, cc;
```

Comment:
what `pc` and `cc` mean? why can't we move these variables outside for loop above and just clear them at every iteration to reduce a number of memory allocations?

Reply:
Plan changed and Current changed. Compiler are very efficient at optimising these allocations. I just tested this in a standalone program with google benchmark. No runtime difference in 10000 repetitions on either clang or gcc.

Reply:
that's actually not true
https://godbolt.org/z/5zTvjo

please consider lines 416, 418 of disassembly